### PR TITLE
Extend the workflow actions to create Docker image

### PR DESCRIPTION
**Pull request by stranljip**
**Motivation**

* for the CD I need a Docker image of the different components

**Modifications**

* add minimal Dockerfile
* add build action for Docker image creation
* This build action needs the two secrets `DOCKER_USERNAME` and `DOCKER_PASSWORD`. I already configured them in the settings for the organization (https://github.com/helmholtz-marketplace)

**Result**

* a Docker image is created and pushed to the Docker Hub

### CODE REVIEW

File: .github/workflows/docker.yml [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+            - name: Set up JDK 11
+              uses: actions/setup-java@v1
+              with:
+                  java-version: 11
+            - name: Build with Maven
+              run: mvn -B package --file pom.xml
+            - name: Push to Docker Hub
+              uses: docker/build-push-action@v1
+              with:
+                  username: ${{ secrets.DOCKER_USERNAME }}
```
Comment (femiadeyemi) [inline .github/workflows/docker.yml:25]: [non-blocking]:
Is this really necessary?

**Review by femiadeyemi — COMMENTED** [review]
My recommendation is that will wait before we go forward with this.

File: .github/workflows/docker.yml (hunk 1 above)
Comment (stranljip) [reply .github/workflows/docker.yml:25]: Yes - I need the updated Docker image on Docker hub in order to recognize it on the CI server and update the container. The only other option would be to have a private Docker registry which we can use. At the moment I do not see that this is necessary.

**Comment by stranljip** [comment]
> My recommendation is that will wait before we go forward with this.

Why? -> Let's discuss it in the weekly today

**Pull request merged**